includes/Admin.php

```diff
@@ -82,4 +84,87 @@ public static function admin_body_class( $class ) {
 
 		return $class;
 	}
+
+	/**
+	 * Pre-fills post content with a web-story/embed block.
+	 *
+	 * @param string $content Default post content.
+	 *
+	 * @return string Pre-filled post content if applicable, or the default content otherwise.
+	 */
+	public static function prefill_post_content( $content ) {
+		$embed_story = isset( $_GET['from-web-story'] ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+		if ( ! $embed_story ) {
+			return $content;
+		}
+
+		$post_id = absint( $_GET['from-web-story'] ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+		// phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped
+		if ( $post_id && Story_Post_Type::POST_TYPE_SLUG === get_post_type( $post_id ) && current_user_can( 'read_post', $post_id ) ) {
+			$url        = (string) get_the_permalink( $post_id );
+			$title      = (string) get_the_title( $post_id );
+			$has_poster = has_post_thumbnail( $post_id );
+
+			ob_start();
```

Comment:
Instead of using ob outputs, why not just use a simple `sprintf`? With placeholders.

Reply:
IMO that would be less readable with such a huge string

Reply:
But it would allow for better escaping not a repeated code in an if statement.

Reply:
The if statement is important because the markup is quite different depending on whether there's a poster image  or not. We should keep that.

However, we could perhaps use `sprintf` together with heredoc